File: cloudflare-worker/src/predictors/strategies/random.js
/**
 * 随机策略
 * 完全随机选择号码
 * 策略：纯随机，不考虑历史数据
 */

import { BaseStrategy } from './base.js';

export class RandomStrategy extends BaseStrategy {
  constructor() {
    super(
      '随机策略',
      '完全随机选择号码，不考虑历史数据'
    );
  }

  /**
   * 生成红球组合
   */
  generateRedBalls(context) {
    const balls = this.randomSelect(this.RED_RANGE, 6);
    
    // 验证组合有效性
    if (!this.isValidRedCombination(balls)) {
      return this.generateRedBalls(context);
    }
    
    return balls.sort();
  }

  /**
   * 生成蓝球
   */
  generateBlueBall(context) {
    return this.BLUE_RANGE[Math.floor(Math.random() * this.BLUE_RANGE.length)];
  }
}

File: lotteries/qlc/strategies/random.py
"""
随机策略 - 完全随机选择
"""

from .base import BaseStrategy
from typing import List, Dict, Tuple
import random


class RandomStrategy(BaseStrategy):
    """随机策略：完全随机选择号码"""
    
    def __init__(self):
        super().__init__(
            name='随机策略',
            description='完全随机选择号码，不考虑历史数据'
        )
    
    def generate_balls(self, context: Dict) -> Tuple[List[int], int]:
        """生成基本号和特别号"""
        # 随机选择7个基本号
        basic_balls = self.random_select(self.BASIC_RANGE, self.BASIC_COUNT)
        
        # 从剩余号码中随机选择特别号
        available_for_special = [b for b in self.BASIC_RANGE if b not in basic_balls]
        special_ball = random.choice(available_for_special) if available_for_special else random.choice(self.BASIC_RANGE)
        
        return sorted(basic_balls), special_ball
